### DISCUSSION TITLE
Refresh materialized view

### DISCUSSION THREAD

**Discussion by mbto** [ClickHouse/ClickHouse#31243] [Q&A]
Hi. I found a pull request https://github.com/ClickHouse/ClickHouse/pull/24921  Materialized view alter refresh and periodic refresh
It's on the 2021 roadmap https://github.com/ClickHouse/ClickHouse/issues/17623
```
Refreshable materialized views and cron jobs.
In progress.
```
What is the current way to update the materialized view if data has changed in the source table?

**Comment by leticiawebb** [comment]
@mbto running through open questions and I noticed this one has been here a while. Please comment if this is still an issue. If I don't hear back by June 1, I will assume you were able to find a resolution. Happy to help if this is still an issue.

**Reply by mbto** [reply]
Yeah, we need a refreshable materialized views

**Reply by leticiawebb** [reply]
I looked into the original pull request and it has been superseded by #33919 . There is a [proposed interim solution](https://github.com/ClickHouse/ClickHouse/issues/33919#issuecomment-1498463545) utilizing external cron jobs to refresh the table using ClickHouse local. Please take a look to see if the interim solution will work for now and please comment if there are additional considerations for the feature.
